### ISSUE TITLE
Findbar - how to move close button to findbars left?

### ISSUE THREAD

**Issue by gabri25ele** [Aris-t2/CustomCSSforFx#70]
is possible move the findbar closebutton to the left?
early I used this:

.findbar-closebutton {
 -moz-box-ordinal-group: 0 !important;
  margin-left: 12px !important;
  margin-right: -4px !important; }

but now with CustomCSSforFx i don't know where to insert.

**Comment by Aris-t2** [comment]
You can not use this code with this projects default **short findbar on top**. Either disable `@import url(./css/generalui/findbar_on_top.css);` in _userChrome.css_ and use this code at the bottom of _userChrome.css_:

```css
:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  -moz-box-ordinal-group: 0 !important;
}
```

Or use this code, if you want to keep the findbar on top (it won't be short anymore, but the close icon will be on the left).

```css
:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) {
  position: static !important;
  border: 0px solid transparent !important;
  border-bottom: 1px solid #e1e1e2 !important;
}

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  -moz-box-ordinal-group: 0 !important;
}
```

**Comment by gabri25ele** [comment]
ok, thank you...
but, in all the case I ever have the findbar  on bottom

https://s17.postimg.org/n5ncul1mn/Cattura.png

**Comment by Aris-t2** [comment]
Add the given code at the bottom of userChrome.css not inside General UI area.
```css
:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  -moz-box-ordinal-group: 0 !important;
}

```
Add a `/*` before `@import url(./css/generalui/findbar_on_top.css);` to disable it.

Use **Notepad++** to view CSS files. Windows default **Notepad** does not highlight code and makes it harder to see what to do.

**Comment by gabri25ele** [comment]
ok very thank you, now is almost perfect 👍 
the close button is too much close to margin window
I think is possibile to insert also "margin-left: 12px !important;", right?

**Comment by Aris-t2** [comment]
Yes, now you can adjust the button with `margin-left` and `margin-right`.

**Comment by userOperaFF** [comment]
```css
/* Findbar on top, short and X on left - experimental */ 

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-container {
  display: inline-block !important;
  float: right !important;
}

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  display: inline-block !important;
  float: left !important;
}
```

![obraz](https://user-images.githubusercontent.com/33114475/33242090-989819a2-d2cf-11e7-83d0-a088d989da8a.png)



⚠️  `X` is vertical align on top ⚠️

**Comment by userOperaFF** [comment]
```css
/* Findbar on top, short and X on left - experimental */ 

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-container {
  display: inline-block !important;
  float: right !important;
}

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  display: inline-block !important;
  float: left !important;
}
```

![obraz](https://user-images.githubusercontent.com/33114475/33242090-989819a2-d2cf-11e7-83d0-a088d989da8a.png)



⚠️  `X` is vertical align on top ⚠️

**Comment by userOperaFF** [comment]
```css
/* Findbar on top, short and X on left - experimental */ 

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-container {
  display: inline-block !important;
  float: right !important;
}

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  display: inline-block !important;
  float: left !important;
}
```

![obraz](https://user-images.githubusercontent.com/33114475/33242090-989819a2-d2cf-11e7-83d0-a088d989da8a.png)



⚠️  `X` is vertical align on top ⚠️  - I can not find good solution fix it.

**Comment by gabri25ele** [comment]
this is my findbar
https://i.imgur.com/YRIdogc.png

this is my lines in userChrome.css:

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) {
  position: static !important;
  border: 0px solid transparent !important;
  margin-left: 12px !important;
  border-bottom: 1px solid #e1e1e2 !important;
}

:-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) .findbar-closebutton {
  -moz-box-ordinal-group: 0 !important;
}

the  only "little defect" is the different color in the little space at the beginning of the bar:
https://i.imgur.com/YRIdogc.png

I do not know who the fault is... ;)

**Comment by Aris-t2** [comment]
Test the options I've added in latest update.

**Comment by userOperaFF** [comment]
`X` no have lw-theme :disappointed:

**Comment by Aris-t2** [comment]
Look like this for me only using 1.3.8 files. Tested a few lw-themes.

Findbars top position + close button at findbars start
![01](https://user-images.githubusercontent.com/6013569/33273591-09a161b2-d38e-11e7-946d-c869ae8b0c48.png)

Findbars bottom position + close button at findbars start
![02](https://user-images.githubusercontent.com/6013569/33273598-0e20c1d8-d38e-11e7-8eec-208b1d0fcfdd.png)

**Comment by gabri25ele** [comment]
With the 1.3.8 files the findbar go over the page (with hidden page items):
<a href="https://postimages.org/" target="_blank"><img src="https://s17.postimg.org/tsaze51f3/image.png" alt="image"/></a><br/><br/>

with my edit file the findbar don't cover the page:
<a href="https://postimages.org/" target="_blank"><img src="https://s17.postimg.org/wz5ixrtkv/image.png" alt="image"/></a><br/><br/>

**Comment by userOperaFF** [comment]
`findbar_on_top_close_at_findbars_start.css` works ok.

👍 

![obraz](https://user-images.githubusercontent.com/33114475/33279907-97406ed4-d3a0-11e7-8fef-9dcf9f546279.png)

**Comment by userOperaFF** [comment]
Bug in Windows 10 (strange white opacity):

![obraz](https://user-images.githubusercontent.com/33114475/33280774-7af0fade-d3a3-11e7-8c94-8d37b9c44ccc.png)

**Comment by Aris-t2** [comment]
@gabri25ele 
Pulling down webcontent is not an option for me.

@userOperaFF 
Will be fixed on next update.

**Comment by userOperaFF** [comment]
New glitch on Windows 7 with Classic system theme with dpi 125%.
After update to 1.3.9 version.

|2/3 px opacity below border bottom: | ![image](https://user-images.githubusercontent.com/33114475/33308782-3154a2b2-d41c-11e7-9781-f8bc19bc753a.png)|
|:---:|:---:|
| **Zoom 200% image:** | ![image](https://user-images.githubusercontent.com/33114475/33308829-5ec24736-d41c-11e7-9fe6-0451e5340ca4.png "Zoom 200% image")|




I do not think so guilty:
```css
/* separator visible bug repair - https://bugzilla.mozilla.org/show_bug.cgi?id=1389633 */ 
scrollbox toolbarseparator {
  -moz-appearance: separator;
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-left: 3px solid transparent;
  border-right: 3px solid transparent;
  -moz-border-left-colors  : transparent transparent ThreeDShadow;
  -moz-border-right-colors : transparent transparent ThreeDHighlight;
}

/* fixed theme background in Maximized (Windows 7, DPI 125%, Full HD) */
#main-window:not([style*='--lwt-header-image:url("resource:///chrome/browser/content/browser/defaultthemes/compact.header.png");']):-moz-lwtheme :-moz-any(.browserContainer,#viewSource) :-moz-any(findbar,#FindToolbar) {
  background-position: -5px -293px !important;
}
```

**Comment by Aris-t2** [comment]
@userOperaFF 

Was this bug present in 1.3.8 as well?
Is this bug also present on Windows 8/10?
Is the bug present, if "close icon" is not on the left? 

What exactly the toolbarseparator glitch with Windows Classic theme (which is a Firefox bug) has to do with findbars "close icon" or did you post the code for information purposes?

**Comment by userOperaFF** [comment]
@Aris-t2 

Tested, fixed 1.4.1 fog :+1:
